src/librbd/watcher/Types.cc

```diff
@@ -29,4 +32,7 @@ void C_NotifyAck::finish(int r) {
 }
 
 } // namespace watcher
-} // namespace librbd
+} // namespace librbdA
+
+template struct librbd::watcher::HandlePayloadVisitor<
```

Comment:
Unnecessary -- the template's definition is available within the translation unit.